lib/Partitioner/Partitioner.cpp

```diff
@@ -723,30 +542,20 @@ llvm::Error Partitioner::QuantizationProfilingPartition(
         << "Profiling a model to be partitioned cross different backends. Each "
            "sub-network will be optimized and run on cpu backend.\n";
   }
-  return llvm::Error::success();
+  return partitions;
```

Comment:
while refactoring can you replace asserts with DCHECK and add more detail to error message where applicable.